packages/core-flows/src/payment/steps/refund-payment.ts

```diff
@@ -0,0 +1,23 @@
+import { ModuleRegistrationName } from "@medusajs/modules-sdk"
+import { IPaymentModuleService } from "@medusajs/types"
+import { StepResponse, createStep } from "@medusajs/workflows-sdk"
+
+type StepInput = {
+  payment_id: string
+  created_by?: string
+  amount?: number
+}
+
+export const refundPaymentStepId = "refund-payment-step"
+export const refundPaymentStep = createStep(
+  refundPaymentStepId,
+  async (input: StepInput, { container }) => {
+    const paymentModule = container.resolve<IPaymentModuleService>(
+      ModuleRegistrationName.PAYMENT
+    )
+
+    const payment = await paymentModule.refundPayment(input)
+
+    return new StepResponse(payment)
+  }
+)
```

Comment:
q: should we add a compensation in case this step is used elsewhere and the workflow fails?

Reply:
I don't see how we can compensate anything in this case. We can't revert a refund at the third-party provider.

One idea I had was to update the payment status to be `requires_action` or something similar to indicate that the current state of things is incorrect.